src/transactions.rs

```diff
@@ -259,23 +279,54 @@ impl<L: Ledger + Serialize + DeserializeOwned> Transactions<L> {
         })
     }
 
-    /// Iterate through the assets.
+    /// Reload from disc to, rebuilds the indices
+    pub fn reload(&mut self) {
+        self.txn_by_hash = self
+            .store
+            .iter()
+            .filter(|txn| txn.hash.is_some())
+            .map(|txn| (txn.hash.as_ref().unwrap().clone(), txn.uid().clone()))
+            .collect();
+        self.expiring_txns = BTreeMap::new();
+        self.uids_awaiting_memos = HashMap::new();
+        for txn in self.store.iter() {
+            if let Some(timeout) = txn.timeout() {
+                self.expiring_txns
+                    .entry(timeout)
+                    .or_insert_with(HashSet::default)
+                    .insert(txn.uid().clone());
+            }
+            for pending in &txn.pending_uids {
+                self.uids_awaiting_memos.insert(*pending, txn.uid().clone());
+            }
+        }
+    }
+
+    pub fn store(
+        &mut self,
+        uid: &TransactionUID<L>,
+        txn: &Transaction<L>,
+    ) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.store(uid, txn)?)
+    }
```

Comment:
This was for `TransactionEditor` accessing the underlying store so it can save changes to the key value storage.  It should have been private.  I am so still used to C++ style privacy where private fields are only accessible from within their class I made this public on instinct.